src/net_processing.cpp

```diff
@@ -3522,7 +3522,7 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         // to users' AddrMan and later request them by sending getaddr messages.
         // Making nodes which are behind NAT and can only make outgoing connections ignore
         // the getaddr message mitigates the attack.
-        if (!pfrom.IsInboundConn()) {
+        if (!pfrom.IsInboundConn() || !pfrom.RelayAddrsWithConn()) {
```

Comment:
@ariard The way I understood this is: before this PR, the node would not announce it's intention to not participate in addr-relay to the outbound-block-relay peer, so it also respected `getaddr` from the peer. Now, after this PR, the node announces it's intention to not participate in addr-relay. Since the objective of the change is to eventually save memory on (`m_tx_relay`, `m_addr_known`), we can no longer serve the `getaddr` call.

I am not sure I understand how the new connection type should already be excluded as:
```
!pfrom.IsInboundConn() => ( !INBOUND && !INBOUND_BLOCK_RELAY )
( !pfrom.IsInboundConn() || !pfrom.RelayAddrsWithConn() ) => ( !INBOUND || BLOCK_RELAY || INBOUND_BLOCK_RELAY )
```

Reply:
You're right that's an OR! Maybe we can rename `RelayAddrsWithConn()` to `IsFullConn` to reduce confusion, like other predicates on connection types. We're stretching slowly from the #19316 original goal of simplifying handling of connection types...

Reply:
@ariard Agreed; this is the motivation for #20729. Feel free to suggest naming improvements to the scripted diff there.